src/main/java/teammates/ui/webapi/GetFeedbackSessionAction.java

```diff
@@ -71,4 +69,39 @@ public JsonResult execute() {
 
         return new JsonResult(response);
     }
+
+    private FeedbackSessionData getFeedbackSessionDataWithFilteredDeadlines(Intent intent) {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String feedbackSessionName = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+        FeedbackSessionAttributes feedbackSessionAttributes = getNonNullFeedbackSession(feedbackSessionName, courseId);
+
+        FeedbackSessionData response;
+        switch (intent) {
+        case STUDENT_SUBMISSION:
+            // fall-through
+        case STUDENT_RESULT:
+            StudentAttributes studentAttributes = getStudentOfCourseFromRequest(courseId);
```

Comment:
I'm a little worried here that this could lead to extra database reads, especially when getting a feedback session is something that happens quite frequently. But at the same time, it seems necessary to hide everyone else's emails/deadlines - resolving this might be somewhat tricky

Reply:
Should the frontend send this instead? I assume such information is always in the page

Reply:
It sounds possible, but could it be open to spoofing, e.g. sending someone else's email to get their deadline? It makes me think that backend checks should still be done somehow, I'm also not sure if it works with google IDs

Reply:
Not sure how sensitive the `FeedbackResponseDeadline` info is. Though we can think it through about Frontend sending email info while protecting against spoofing.

Reply:
From the frontend's session submission page, we can get the `personEmail` -- if we want to send this over, the frontend can no longer load the person details and the feedback session in parallel, but has to wait for the student/instructor's details to finish loading before loading the feedback session; so there could be some cost involved

Reply:
Just checked the maintainer logs, and from there I'd estimate that `GetFeedbackSessionAction` is called about 4400 times yesterday - assuming it's a typical day in a year (though realistically it probably isn't), we'll have ~1.6m calls/year. IIRC it costs ~$0.04/100k entity reads, so +1 read will amount to <$1 a year. Huh that doesn't sound as bad as I thought it'd be, even though these are just estimates.

The above approach (frontend sending email to backend) also sounds doable, though it seems a bit weird for the frontend to be pulling the email from the backend, so that it can send the same info right back to it

Reply:
It is to respect the RESTful API unfortunately, though this is really just front end caching of DB data. Technically, if we have DB cache, this won't be a problem